### PR TITLE
Replace redundant resource type and switch to readytorun output

### PR DESCRIPTION
**Pull request by jrabbott**
Replace redundant resource type and switch to ReadyToRun output

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service_plan
https://learn.microsoft.com/en-us/dotnet/core/deploying/ready-to-run

### CODE REVIEW

File: web/terraform/main.tf [hunk 1]
```diff
@@ -53,23 +53,20 @@ resource "azurerm_resource_group" "resource-group" {
   tags     = local.common-tags
 }
 
-resource "azurerm_app_service_plan" "education-benchmarking-asp" {
+resource "azurerm_service_plan" "education-benchmarking-asp" {
```
Comment (WolfyUK) [inline web/terraform/main.tf:56]: Will destroying the old service plan affect the dependent app service?

**Comment by WolfyUK** [comment]
I guess once the app service plan is updated we can remove the pinned `azurerm` provider version number (see #82)?

File: web/terraform/main.tf (hunk 1 above)
Comment (jrabbott) [reply web/terraform/main.tf:56]: Yes, there will need to be some manual intervention as there is a "clean" migration path,

**Comment by jrabbott** [comment]
> I guess once the app service plan is updated we can remove the pinned `azurerm` provider version number (see #82)?



> I guess once the app service plan is updated we can remove the pinned `azurerm` provider version number (see #82)?

I'm not sure the 2 are related.

**Comment by WolfyUK** [comment]
> > I guess once the app service plan is updated we can remove the pinned `azurerm` provider version number (see #82)?
> 
> I'm not sure the 2 are related.

@michael-fielding Was this the reason for pinning the provider version last week, or was it something else?

**Comment by michael-fielding** [comment]
> I guess once the app service plan is updated we can remove the pinned `azurerm` provider version number (see #82)?

**Comment by michael-fielding** [comment]
> > > I guess once the app service plan is updated we can remove the pinned `azurerm` provider version number (see #82)?
> > 
> > 
> > I'm not sure the 2 are related.
> 
> @michael-fielding Was this the reason for pinning the provider version last week, or was it something else?

No, the reason for pinning the version was the bug introduced when Terraform started enforcing case-sensitivity in the "serverFarms" part of the app-service path

**Review by michael-fielding — APPROVED** [review]

**Pull request merged**